File: leetcode/206. Reverse Linked List/206._Reverse_Linked_List_2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode current = head, last = null, next;
        while( current != null) {
            next = current.next;
            current.next = last;
            last = current;
            current = next;
        }
        return last;
    }
}

File: leetcode/206. Reverse Linked List/206._Reverse_Linked_List_2.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        last = None
        current = head
        while current:
            hold = current.next
            current.next = last
            last = current
            current = hold
        return last